src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+        } else {
+            this->player.actor.home.rot.y = this->player.actor.shape.rot.y + 0x8000;
+        }
+        this->player.stateFlags2 |= PLAYER_STATE2_40000;
+        play->startPlayerCutscene(play, &this->player, -1);
+    }
+    this->unk_D78 = D_80A418A8[arg2->unk_1_4];
+    return true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3FE20.s")
+s32 func_80A3F73C(EnTest3* this, PlayState* play) {
+    if (Actor_ProcessTalkRequest(&this->player.actor, &play->state)) {
+        func_80A3E7E0(this, func_80A4084C);
+        this->player.unk_730 = &GET_PLAYER(play)->actor;
+        this->player.stateFlags2 &= ~PLAYER_STATE2_40000;
+        D_80A41D5C = true;
+        if ((this->unk_D78->unk_0 == 4) && (gSaveContext.save.weekEventReg[51] & 8)) {
+            func_80151BB4(play, 2);
+        }
+    } else {
+        if (play->actorCtx.unk5 & 0x10) {
+            play->actorCtx.unk5 &= ~0x10;
+            this->player.stateFlags2 &= ~PLAYER_STATE2_40000;
+            this->unk_D90->stateFlags1 |= PLAYER_STATE1_20;
+            func_800BC154(play, &play->actorCtx, &this->unk_D90->actor, 4);
+            func_800BC154(play, &play->actorCtx, &this->player.actor, 2);
+            ActorCutscene_SetReturnCamera(this->unk_D8E);
+            play->startPlayerCutscene(play, &this->player, 7);
+        }
+        func_800B863C(&this->player.actor, play);
+        if (this->unk_D88 == 3) {
+            func_80A3F534(this, play);
+        } else if (this->unk_D88 == 5) {
+            func_80A3F5A4(this, play);
+        }
+        this->player.actor.textId = this->unk_D78->textId;
+        this->player.actor.flags |= (ACTOR_FLAG_8 | ACTOR_FLAG_1);
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3FF10.s")
+s32 func_80A3F8D4(EnTest3* this, PlayState* play, struct_80A41828* arg2, ScheduleOutput* arg3) {
+    Actor* postActor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3FFD0.s")
+    func_80A3F15C(this, play, arg2);
+    if (((postActor = func_80A3F2BC(play, this, ACTOR_EN_PST, ACTORCAT_PROP, 100.0f, 20.0f)) != NULL) ||
+        ((postActor = func_80A3F2BC(play, this, ACTOR_EN_PM, ACTORCAT_NPC, 100.0f, 20.0f)) != NULL)) {
+        this->player.actor.home.rot.y = Actor_YawBetweenActors(&this->player.actor, postActor);
+    }
+    play->startPlayerCutscene(play, &this->player, 0x61);
+    return true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40098.s")
+s32 func_80A3F9A4(EnTest3* this, PlayState* play) {
+    Math_ScaledStepToS(&this->player.actor.shape.rot.y, this->player.actor.home.rot.y, 0x320);
+    this->player.currentYaw = this->player.actor.shape.rot.y;
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40230.s")
+s32 func_80A3F9E4(EnTest3* this, PlayState* play, struct_80A41828* arg2, ScheduleOutput* arg3) {
+    arg3->time0 = (u16)(gSaveContext.save.time - 0x3FFC);
```

Comment:
```suggestion
    arg3->time0 = SCHEDULE_TIME_NOW;
```
Not sure if the case is still needed.

Reply:
The cast is still needed. However, the cast can be made part of `SCHEDULE_TIME_NOW` without breaking anything. The game seems to really want that to be a u16.